src/env.rs

```diff
@@ -195,11 +216,22 @@ impl Environment {
             data_device_manager,
             decorations_mgr,
             outputs,
+            surfaces,
         })
     }
 
     /// Retrieve the accepted SHM formats of the server
     pub fn shm_formats(&self) -> Vec<wl_shm::Format> {
         self.shm_formats.lock().unwrap().clone()
     }
+
+    /// Create a new dpi aware surface
+    pub fn create_surface<F>(&self, dpi_change: Box<F>) -> Proxy<wl_surface::WlSurface>
```

Comment:
Please do the boxing of the callback internally, and let the public method take non-boxed closure.